src/Nethermind/Nethermind.Db/Blooms/FixedSizeFileStore.cs

```diff
@@ -58,47 +51,19 @@ public void Write(long index, ReadOnlySpan<byte> element)
 
         public int Read(long index, Span<byte> element)
         {
-            EnsureFlushed();
-
-            lock (_fileRead)
-            {
-                SeekIndex(_fileRead, index);
-                return _fileRead.Read(element);
-            }
+            return RandomAccess.Read(_file, element, GetPosition(index));
         }
 
         public IFileReader CreateFileReader()
         {
-            EnsureFlushed();
             return new FileReader(_path, _elementSize);
         }
 
-        private void EnsureFlushed()
-        {
-            if (Interlocked.CompareExchange(ref _needsFlush, 0, 1) == 1)
-            {
-                lock (_fileWrite)
-                {
-                    _fileWrite.Flush();
```

Comment:
Removing of this might change the behavior, as AFAIK `RandomAccess` won't issue any `FlushFileBuffers` on Win or similar one one `Linux` (if there's any). Again, this depends on the usage of the API, whether it's concurrent or not and whether the file should be persisted atomically or not (_atomically_ is not the best word to use here but the best that I came up with).

Reply:
Flush was never called with `Flush(flushToDisk: true)` so it was only flushing its internal memory buffers, effect would be the same. As reads and writes are essentially random access and always seek before read and write the internal buffers aren't much help.

I did add a manual flushToDisk and it pushed my SSD to 99%; so it defiantly wasn't doing that before, so took it back out